optional_plugins/varianter_yaml_to_mux/avocado_varianter_yaml_to_mux/__init__.py

```diff
@@ -314,10 +314,60 @@ def configure(self, parser):
                              help="DEPRECATED: Filter out path(s) from "
                              "multiplexing (use --mux-filter-out instead)")
 
+            mux = subparser.add_argument_group("yaml to mux testsuite options")
+            mux.add_argument("--mux-suite-only", nargs="+",
+                             help="Filter only part of the YAML suite file")
+            mux.add_argument("--mux-suite-out", nargs="+",
+                             help="Filter out part of the YAML suite file")
+
     def run(self, args):
         """
         The YamlToMux varianter plugin handles these
         """
+        loader.loader.register_plugin(YamlTestsuiteLoader)
+
+
+class YamlTestsuiteLoader(loader.TestLoader):
+
+    name = "yaml_testsuite"
+
+    @staticmethod
+    def get_type_label_mapping():
+        """
+        This needs to be staticmethod but we only know the mapping
+        after parsing the provided files. Let's accept only FileLoader
+        type for now and when we have way to report mapping after
+        initialization add support to specify type in YAML file
+        via __import__(suite.get("loader_type")).discover(...)
+        """
+        return loader.FileLoader.get_type_label_mapping()
+
+    @staticmethod
+    def get_decorator_mapping():
+        return loader.FileLoader.get_decorator_mapping()
+
+    def discover(self, reference, which_tests=loader.DEFAULT):
+        tests = []
+        try:
+            root = mux.apply_filters(create_from_yaml([reference], False),
+                                     getattr(self.args, "mux_suite_only", []),
+                                     getattr(self.args, "mux_suite_out", []))
+        except Exception:
+            return []
+        mux_tree = mux.MuxTree(root)
+        # TODO: Add __import__(suite.get("loader_type")).discover(...)
+        # support to allow non-FileLoader tests. Currently not possible
+        # due to get_type_label_mapping (details viz it's docstring)
+        f_loader = loader.FileLoader(self.args, {})
+        for variant in mux_tree:
+            params = varianter.AvocadoParams(variant, "YamlTestsuiteLoader",
+                                             ["/run/*"], {})
+            reference = params.get("test_reference")
+            _tests = f_loader.discover(reference)
+            for tst in _tests:
+                tst[1]["params"] = (variant, ["/run/*"])
```

Comment:
This is kind of interesting.  The fact that you set a `params` argument to the test class, triggered some legacy code in the list plugin, which causes a crash:

```
Traceback (most recent call last):
  File "/home/cleber/.local/bin/avocado", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "/home/cleber/src/avocado/avocado/scripts/avocado", line 86, in <module>
    sys.exit(app.run())
  File "/home/cleber/src/avocado/avocado/avocado/core/app.py", line 90, in run
    return method(self.parser.args)
  File "/home/cleber/src/avocado/avocado/avocado/plugins/list.py", line 193, in run
    return test_lister.list()
  File "/home/cleber/src/avocado/avocado/avocado/plugins/list.py", line 137, in list
    self._list()
  File "/home/cleber/src/avocado/avocado/avocado/plugins/list.py", line 132, in _list
    test_matrix, stats, tag_stats = self._get_test_matrix(test_suite)
  File "/home/cleber/src/avocado/avocado/avocado/plugins/list.py", line 68, in _get_test_matrix
    id_label = params['params']['id']
TypeError: tuple indices must be integers, not str
```

I dug deeper and confirmed that was legacy behavior, and that your new `parameter` argument triggered a mistaken interpretation of what `params` would contain.  The fix is here:

https://github.com/avocado-framework/avocado/pull/2120

With this fix applied, listing works as expected:

```
$ avocado list examples/yaml_to_mux_loader/basic.yaml 
INSTRUMENTED passtest.py:PassTest.test
INSTRUMENTED sleeptest.py:SleepTest.test
INSTRUMENTED failtest.py:FailTest.test
INSTRUMENTED failtest.py:FailTest.test
INSTRUMENTED failtest.py:FailTest.test
```